src/fastga/models/aerodynamics/components/digitization/compute_cn_delta_a_correlation_constant.py

```diff
@@ -0,0 +1,152 @@
+    # Overriding OpenMDAO setup_partials
+    def setup_partials(self):
+        self.declare_partials(of="aileron_correlation_constant", wrt="*", method="exact")
+
+    # pylint: disable=missing-function-docstring, unused-argument
+    # Overriding OpenMDAO compute, not all arguments are used
+    def compute(self, inputs, outputs, discrete_inputs=None, discrete_outputs=None):
+        taper_ratio = inputs["data:geometry:wing:taper_ratio"]
+        aspect_ratio = inputs["data:geometry:wing:aspect_ratio"]
+        aileron_inner_span_ratio = 1.0 - inputs["data:geometry:wing:aileron:span_ratio"]
+
+        if taper_ratio != np.clip(taper_ratio, 0.25, 1.0):
+            taper_ratio = np.clip(taper_ratio, 0.25, 1.0)
+            _LOGGER.warning("Taper ratio is outside of the range in Roskam's book, value clipped")
+
+        if aspect_ratio != np.clip(aspect_ratio, 3.0, 8.0):
+            aspect_ratio = np.clip(aspect_ratio, 3.0, 8.0)
+            _LOGGER.warning("Aspect ratio is outside of the range in Roskam's book, value clipped")
+
+        if aileron_inner_span_ratio != np.clip(aileron_inner_span_ratio, 0.0, 0.91):
+            aileron_inner_span_ratio = np.clip(aileron_inner_span_ratio, 0.0, 0.91)
+            _LOGGER.warning(
+                "Aileron inboard location is outside of the range in Roskam's book, value clipped"
+            )
+
+        outputs["aileron_correlation_constant"] = (
+            -0.63632822
+            - 0.57304492 * taper_ratio
+            + 0.21707743 * aspect_ratio
+            + 0.10407542 * aileron_inner_span_ratio
+            + 0.61587892 * taper_ratio**2.0
+            + 0.04971284 * taper_ratio * aspect_ratio
+            - 0.20067475 * taper_ratio * aileron_inner_span_ratio
+            - 0.02997066 * aspect_ratio**2.0
+            - 0.00550507 * aspect_ratio * aileron_inner_span_ratio
+            - 0.00622554 * aileron_inner_span_ratio**2.0
+            - 0.23509612 * taper_ratio**3.0
+            - 0.02728276 * taper_ratio**2.0 * aspect_ratio
+            + 0.12690424 * taper_ratio**2.0 * aileron_inner_span_ratio
+            - 0.00050367 * taper_ratio * aspect_ratio**2.0
+            - 0.01379183 * taper_ratio * aspect_ratio * aileron_inner_span_ratio
+            + 0.04289006 * taper_ratio * aileron_inner_span_ratio**2.0
+            + 0.00140254 * aspect_ratio**3.0
+            + 0.00149011 * aspect_ratio**2.0 * aileron_inner_span_ratio
+            - 0.00553262 * aspect_ratio * aileron_inner_span_ratio**2.0
+            - 0.06766704 * aileron_inner_span_ratio**3.0
+        )
+
+    # pylint: disable=missing-function-docstring, unused-argument
+    # Overriding OpenMDAO compute_partials, not all arguments are used
+    def compute_partials(self, inputs, partials, discrete_inputs=None):
+        taper_ratio = inputs["data:geometry:wing:taper_ratio"]
+        aspect_ratio = inputs["data:geometry:wing:aspect_ratio"]
+        span_ratio = inputs["data:geometry:wing:aileron:span_ratio"]
```

Comment:
Wait why are you using spanratio here whereas in the compute you use 1.0 - span ratio ? Do the partials computation take that into account ?